src/graphics.rs

```diff
@@ -1628,6 +1634,263 @@ impl Drawable for Sprite {
+                texture_height as u16,
+                &texture,
+            )),
+        })
+    }
+
+    fn get_char_list(char_list_type: CharacterListType) -> Result<Vec<char>> {
+        let char_list = match char_list_type {
+            CharacterListType::AsciiSubset => [0x20..0x7F].iter(),
+            CharacterListType::Ascii => [0x00..0xFF].iter(),
+            CharacterListType::Cyrillic => [
+                0x0020u32..0x00FF, // Basic Latin + Latin Supplement
+                0x0400u32..0x052F, // Cyrillic + Cyrillic Supplement
+                0x2DE0u32..0x2DFF, // Cyrillic Extended-A
+                0xA640u32..0xA69F, // Cyrillic Extended-B
+            ]
+            .iter(),
+            CharacterListType::Thai => [
+                0x0020u32..0x00FF, // Basic Latin
+                0x2010u32..0x205E, // Punctuations
+                0x0E00u32..0x0E7F, // Thai
+            ]
+            .iter(),
+
+            CharacterListType::Vietnamese => [
+                0x0020u32..0x00FF, // Basic Latin
+                0x0102u32..0x0103,
+                0x0110u32..0x0111,
+                0x0128u32..0x0129,
+                0x0168u32..0x0169,
+                0x01A0u32..0x01A1,
+                0x01AFu32..0x01B0,
+                0x1EA0u32..0x1EF9,
+            ]
+            .iter(),
+            CharacterListType::Chinese => return Err(anyhow!("Chinese fonts not yet supported")),
+            CharacterListType::Japanese => return Err(anyhow!("Japanese fonts not yet supported")),
+        };
+        char_list
+            .cloned()
+            .flatten()
+            .map(|c| {
+                std::char::from_u32(c).ok_or(anyhow!("Unable to convert u32 \"{}\" into char", c))
+            })
+            .collect::<Result<Vec<char>>>()
+    }
+}
+
+impl Drawable for FontAtlas {
+    fn draw(&self, ctx: &mut Graphics, instance: InstanceParam) {
+        self.font_texture.load().draw(ctx, instance);
+    }
+
+    fn aabb(&self) -> Box2<f32> {
+        self.font_texture.load().aabb()
+    }
+}
+
+pub struct Text {
+    batch: SpriteBatch,
+    char_texture_map: HashMap<char, CharInfo>,
+}
+
+impl Text {
+    pub fn new(ctx: &mut Graphics, input_text: &str, font_atlas: &FontAtlas, color: Color) -> Self {
+        let mut text = Text {
+            batch: SpriteBatch::with_capacity(ctx, font_atlas.font_texture.clone(), input_text.len()),
+            char_texture_map: font_atlas.font_map.clone(),
+        };
+        text.set_text(input_text, color);
+        text
+    }
+
+    pub fn set_text(&mut self, new_text: &str, color: Color) {
+        self.batch.clear();
+        let mut width: f32 = 0.;
+        for c in new_text.chars() {
+            let c_info = self.char_texture_map.get(&c).unwrap_or(self.char_texture_map.get(&'a').unwrap());
+            let mut i_param = InstanceParam::new().src(c_info.uvs).translate2(Vector2::new(width, c_info.vertical_offset as f32));
+            i_param.color = color;
```

Comment:
This seems like a good reason to add a `.color(color)` method to `InstanceParam` with the type signature `fn(self, Color) -> Self` similar to `translate2` and `src`. Weird to have it be `mut` just to set one field.